components/formats-gpl/src/loci/formats/in/ColumbusReader.java

```diff
@@ -0,0 +1,737 @@
+        }
+        else if (name.equals("Col")) {
+          p.col = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("FieldID")) {
+          p.field = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("TimepointID")) {
+          p.timepoint = Integer.parseInt(value) - 1;
+          if (p.timepoint == 0) {
+            p.timepoint = externalTime;
+          }
+        }
+        else if (name.equals("ChannelID")) {
+          p.channel = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("ChannelName")) {
+          p.channelName = value;
+        }
+        else if (name.equals("ChannelColor")) {
+          Long color = new Long(value);
+          int blue = (int) ((color >> 24) & 0xff);
+          int green = (int) ((color >> 16) & 0xff);
+          int red = (int) ((color >> 8) & 0xff);
+          int alpha = (int) (color & 0xff);
+
+          // not setting this yet, and deferring to the
+          // emission wavelength to set the color
+          // neither BGRA nor ARGB seems to make sense for all channels
+          //p.channelColor = new Color(red, green, blue, alpha);
+        }
+        else if (name.equals("MeasurementTimeOffset")) {
+          p.deltaT = new Double(value);
+        }
+        else if (name.equals("AbsTime")) {
+          p.deltaT = new Timestamp(value).asInstant().getMillis() / 1000d;
+        }
+        else if (name.equals("MainEmissionWavelength")) {
+          p.emWavelength = new Double(value);
+        }
+        else if (name.equals("MainExcitationWavelength")) {
+          p.exWavelength = new Double(value);
+        }
+        else if (name.equals("ImageResolutionX")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.sizeX = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("ImageResolutionY")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.sizeY = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionX")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionX = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionY")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionY = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionZ")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionZ = correctUnits(new Double(value), unit);
+        }
+      }
+
+      planes.add(p);
+    }
+
+  }
+
+  private Double correctUnits(Double v, String unit) {
+    if (unit == null) {
+      return v;
+    }
+
+    if (unit.equals("m")) {
+      return v * 1000000;
+    }
+    else if (unit.equals("cm")) {
+      return v * 10000;
+    }
+    else if (unit.equals("nm")) {
+      return v /= 1000;
+    }
+    return v;
```

Comment:
They are lowercase in the examples we have. I'm not sure if we know in advance that they will always be lowercase, but they should be in order to be valid units. /cc @melissalinkert